src/App.jsx

```diff
@@ -0,0 +1,506 @@
+import React, { useState, useRef } from "react";
```

Comment:
It would be better to split the component into smaller ones and organize the project structure properly